src/main/scala/org/sunbird/cloud/storage/service/GcloudStorageService.scala

```diff
@@ -85,4 +91,35 @@ class GcloudStorageService(config: StorageConfig) extends BaseStorageService  {
     } else
       throw new StorageServiceException("uri not available for the given prefix: "+ _prefix)
   }
+
+  /**
+   * Method to get V4 Signed URL when storage is GCP
+   * @param re
+   * @return
+   */
+  override def getPutSignedURL(container: String, objectKey: String,  ttl: Option[Int], permission: Option[String] = Option("r"), contentType: Option[String], additionalParams: Option[Map[String,String]] = None): String = {
```

Comment:
The implementation of this method can improve.
1. `additionalParams` - Please try to use case class. Which will explain the parameter clearly. With Map, it is difficult to understand.
2. Don't use Java Map (java.util.HashMap) and Scala Map at single place. Try to use Scala.f
3. Avoid returning `null`. Through exception and expect the required parameters. This will avoid NPE.

Reply:
Please find the comments below
1. additional - We have defined it as map so that client can send additional information if required with minimal change in implementations 
2. Using scala map - done
3. Avoid using null - done